frappe/core/report/transactionlog_report/transactionlog_report.py

```diff
@@ -0,0 +1,203 @@
+
+from __future__ import unicode_literals
+import frappe, json
+from  frappe import _
+import hashlib
+import ast
+
+def execute(filters=None):
+	columns, data = [], []
+
+	columns = getColumn()
+	data = getData(filters)
+
+	return columns, data
+
+
+def getData(filters=None):
+
+	full = full_data()
+	return full
+
+def full_data(): #function to get the Full data
+
+	f_data = frappe.db.sql("SELECT * FROM tabTransactionLog order by creation desc ", as_dict=1)
+	result = []
+	remark = ''
+	for fd in f_data:
+		data = json.loads(fd.data)
+		row_index = int(fd.row_index)
+		if row_index > 1:
+			row_index = int(fd.row_index) - 1
+
+		previous_hash = frappe.db.sql("SELECT chaining_hash FROM `tabTransactionLog` WHERE row_index = {0}".format(row_index))
+		remark = doc_check(fd.reference_doctype, fd.document_name, previous_hash)
+		frappe.logger().debug(remark)
+		if remark:
+			if remark[0]== 'Previous Document Missing':
+				if remark[1]:
+					remark = remark[0]
+		if fd.reference_doctype == 'Sales Invoice':
+			s_invoice = frappe.db.sql("SELECT status, base_paid_amount FROM `tabSales Invoice` where name= '{0}' ".format(fd.document_name),as_dict=1)
+			if s_invoice:
+				paid_amount = s_invoice[0].base_paid_amount
+				status = s_invoice[0].status
+			else:
+				paid_amount = 0
+				status = 'Unpaid'
+			if data['action'] == 'Printed':
+				row = [remark, fd.creation, fd.owner, fd.modified_by, fd.document_name, data['action'],
+					   fd.reference_doctype,
+					   data['customer'], status, data['company'],
+					   data['posting_date'], data['currency'],
+					   paid_amount, data['total_taxes_and_charges'], data['base_net_total'], data['net_total']]
+			else:
+				row = [remark, fd.creation, fd.owner, fd.modified_by, fd.document_name, data['action'], fd.reference_doctype,
+				   data['party'], status, data['company'],
+				   data['posting_date'], data['account_currency'],
+				   paid_amount, data['total_taxes_and_charges'], data['base_net_total'], data['net_total']]
+		else:
+			if data['action'] == 'Printed':
+				row = [remark, fd.creation, fd.owner, fd.modified_by, fd.document_name, data['action'],
+					   fd.reference_doctype, data['party'], data['payment_type'], data['company'],
+					   data['posting_date'], data['paid_from_account_currency'],
+					   data['paid_amount'], 0, data['total_allocated_amount'], data['paid_amount']]
+			else:
+				row = [remark, fd.creation, fd.owner, fd.modified_by, fd.document_name, data['action'], fd.reference_doctype,
+					data['party'],data['payment_type'],  data['company'],
+					data['posting_date'], data['account_currency'],
+					data['paid_amount'], 0, data['total_allocated_amount'], data['paid_amount']]
+
+		result.append(row)
+	return result
+
+
+def doc_check(reference_doctype, doc_name, previous_hash): #Main function to check the document integrity
+
+	remarks = ''
+	if previous_hash:
+		remarks = hash_check(previous_hash)
+		if reference_doctype == 'Sales Invoice':
```

Comment:
Sales Invoice and Payment Entry are part of ERPNext. You cannot use them in frappe code.